libs/util/http.client.js

```diff
@@ -130,8 +131,8 @@ function mergeConfig(config) {
 function doXhr(method, url, headers, data, config, callback) {
     var options, timeout;
 
+    headers = normalizeHeaders(headers, method, config.cors);
```

Comment:
`mergeConfig` doesn't appear to actually merge the config passed in, it just returns a new config object using some of the properties from the passed-in config to initialize the new one. This means that the `cors` property is no longer present when `normalizeHeaders` is called. I moved it up so that I could access the `cors` property before `mergeConfig` wipes it out. Perhaps `mergeConfig` needs to be reworked a bit?